src/utils/details.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+const cryptoHelper = require('./crypto')
+const DATA_DELIMITER = '\n\n'
+
+function _createRequest ({
+  method,
+  headers,
+  data
+}) {
+  const statusLine = 'PSK/1.0 ' + method.toUpperCase() + '\n'
+  const headerLines = Object.keys(headers)
+    .map((k) => k + ': ' + headers[k])
+    .join('\n') + DATA_DELIMITER
+
+  let rawData = data
+  if (!Buffer.isBuffer(data) && typeof data === 'object') {
+    rawData = Buffer.from(JSON.stringify(data), 'utf8')
+  } else if (data === undefined) {
+    rawData = Buffer.from([])
+  }
```

Comment:
I think those should be set manually by the application layer protocol, because the transport layer doesn't care about the data inside the private PSK request.